File: 7kyu/count-the-digit/count-the-digit.go
package main

import (
	"fmt"
)

func NbDig(n int, d int) int {
	count := 0
	for i := 0; i <= n; i++ {
		square := i * i
		if i == 0 && square%10 == d {
			count++
		}

		for square > 0 {
			if square%10 == d {
				count++
			}
			square /= 10
		}
	}
	return count
}

func main() {
	fmt.Println(NbDig(550, 5))  // 213
	fmt.Println(NbDig(5750, 0)) // 4700
}

File: 7kyu/count-the-digit/count-the-digit.py
# def nb_dig(n, d):
#   count = ""
#   for digit in range(n + 1):
#     count += str(digit ** 2)
#   return len(list(filter(lambda digit: digit == str(d), count)))

# def nb_dig(n, d):
#   count = ""
#   for digit in range(n + 1):
#     count += str(digit ** 2)
#   return len([digit for digit in count if digit == str(d)])

def nb_dig(n, d):
  return str([digit ** 2 for digit in range(n + 1)]).count(str(d))

print(nb_dig(5750, 0)) #4700